core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java

```diff
@@ -570,13 +579,16 @@ private Cache cacheWithFlags(long flagBitSet) {
 
    @Override
    public Object visitSizeCommand(InvocationContext ctx, SizeCommand command) throws Throwable {
-      // Use an unwrapped cache to avoid unneeded transformations as we only want a count
-      long size = cacheWithFlags(command.getFlagsBitSet()).keySet().stream().count();
-      if (size > Integer.MAX_VALUE) {
-         return Integer.MAX_VALUE;
+      ClusterPublisherManager managerToUse;
+      if (command.hasAnyFlag(FlagBitSets.CACHE_MODE_LOCAL)) {
+         managerToUse = localClusterPublisherManager;
       } else {
-         return (int) size;
+         managerToUse = clusterPublisherManager;
       }
+
+      return asyncValue(managerToUse.keyReduction(false, null, null, ctx,
```

Comment:
what are these null, null ? would it be better to overload the keyReduction method for lisibility?

Reply:
I personally don't think so, there are too many optional arguments. And they all have different types. The Javadoc clearly states what they are and what happens if they are null.